modules/ppcp-settings/docs/merchant-onboarding-flow.md

```diff
@@ -0,0 +1,263 @@
+# Merchant Onboarding Flow
+
+The onboarding flow is a multi-step wizard that guides merchants through setting up their PayPal integration. The flow is dynamic and adapts based on merchant preferences, store configuration, and country-specific capabilities determined by configuration flags.
+
+## Core Architecture
+
+### Configuration Flags Value Determination
+
+The onboarding configuration flags are determined during the `OnboardingProfile` instantiation in the dependency injection container. Each flag represents a specific capability or business rule that affects the onboarding flow.
+
+**File:** `modules/ppcp-settings/services.php`
+
+```php
+'settings.data.onboarding' => static function (ContainerInterface $container): OnboardingProfile {
+    $can_use_casual_selling = $container->get('settings.casual-selling.eligible');
+    $can_use_vaulting       = $container->has('save-payment-methods.eligible') && $container->get('save-payment-methods.eligible');
+    $can_use_card_payments  = $container->has('card-fields.eligible') && $container->get('card-fields.eligible');
+    $can_use_subscriptions  = $container->has('wc-subscriptions.helper') && $container->get('wc-subscriptions.helper')->plugin_is_active();
+    $should_skip_payment_methods = class_exists('\WC_Payments');
+    $can_use_fastlane = $container->get('axo.eligible');
+    $can_use_pay_later = $container->get('button.helper.messages-apply');
+
+    return new OnboardingProfile(/* ...flags... */);
+},
+```
+
+#### Flag Value Sources
+
+**`can_use_casual_selling`** - Personal account support
```

Comment:
Suggestion: To me this looks a bit like code documentation that should be present in the `.php` file.
Also, the service name is already visible in the code snippet above and describing the implementation logic here seems risky to get outdated as well.

I would keep this flag overview more concise, documenting the "business perspective", e.g.
- `can_use_casual_selling` - Business accounts are supported in all countries; this flag indicates if the store country also supports onboarding using a personal account
- `can_use_vaulting` - Whether the store's country supports vaulting - displays additional onboarding choices  
- `should_skip_payment_methods` - Whether the "branded only" mode is active and the onboarding wizard is reduced
- ...